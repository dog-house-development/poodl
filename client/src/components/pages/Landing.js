import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class Landing extends Component {
    render() {
        return (
            <div className="landing-container">
                <div className="landing-header">
                    <h1>Welcome to</h1>
                    <h1 className="landing-title">Poodl</h1>
                    <p>The member management tool for rural senior centers!</p>
                    <Link to="/" className="button large primary">
                        Learn more
                    </Link>
                </div>
            </div>
        );
    }
}

export default Landing;
