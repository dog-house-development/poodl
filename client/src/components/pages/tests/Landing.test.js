import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import _ from 'lodash';

import { Landing, mapStateToProps } from '../Landing';

configure({ adapter: new Adapter() });

describe('Landing tests', () => {
    let state, props, wrapper, instance;
    const setInstanceAndWrapper = (_props = {}, _state = {}) => {
        state = _.assign(
            {},
            {
                auth: {
                    isAuthenticated: false
                }
            },
            _state
        );
        props = _.assign({}, { history: ['/'] }, _props);
        wrapper = shallow(<Landing {..._.assign({}, props, mapStateToProps(state, props))} />);
        instance = wrapper.instance();
    };
    beforeEach(() => {
        setInstanceAndWrapper();
    });

    describe('render', () => {
        it('should render correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('mapStateToProps', () => {
        it('should map state to props', () => {
            expect(mapStateToProps(state, props)).toEqual({
                isAuthenticated: false
            });
        });
    });

    describe('componentDidMount', () => {
        it('should redirect if admin is authenticated', () => {
            const authenticatedState = {
                auth: {
                    isAuthenticated: true
                }
            };
            setInstanceAndWrapper({}, authenticatedState);
            const newInstanceProps = instance.props.history.concat('/dashboard');
            instance.componentDidMount();
            expect(instance.props.history).toEqual(newInstanceProps);
        });
    });
});
