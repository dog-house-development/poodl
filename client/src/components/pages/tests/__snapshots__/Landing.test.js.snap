// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing tests render should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="landing-header"
      >
        <h1>
          Welcome to
        </h1>
        <h1
          className="landing-title"
        >
          Poodl
        </h1>
        <p>
          The member management tool for rural senior centers!
        </p>
        <Link
          className="button large primary"
          replace={false}
          to="/"
        >
          Learn more
        </Link>
      </div>,
      "className": "landing-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Welcome to
          </h1>,
          <h1
            className="landing-title"
          >
            Poodl
          </h1>,
          <p>
            The member management tool for rural senior centers!
          </p>,
          <Link
            className="button large primary"
            replace={false}
            to="/"
          >
            Learn more
          </Link>,
        ],
        "className": "landing-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to",
          },
          "ref": null,
          "rendered": "Welcome to",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Poodl",
            "className": "landing-title",
          },
          "ref": null,
          "rendered": "Poodl",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The member management tool for rural senior centers!",
          },
          "ref": null,
          "rendered": "The member management tool for rural senior centers!",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Learn more",
            "className": "button large primary",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Learn more",
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="landing-header"
        >
          <h1>
            Welcome to
          </h1>
          <h1
            className="landing-title"
          >
            Poodl
          </h1>
          <p>
            The member management tool for rural senior centers!
          </p>
          <Link
            className="button large primary"
            replace={false}
            to="/"
          >
            Learn more
          </Link>
        </div>,
        "className": "landing-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Welcome to
            </h1>,
            <h1
              className="landing-title"
            >
              Poodl
            </h1>,
            <p>
              The member management tool for rural senior centers!
            </p>,
            <Link
              className="button large primary"
              replace={false}
              to="/"
            >
              Learn more
            </Link>,
          ],
          "className": "landing-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to",
            },
            "ref": null,
            "rendered": "Welcome to",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Poodl",
              "className": "landing-title",
            },
            "ref": null,
            "rendered": "Poodl",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "The member management tool for rural senior centers!",
            },
            "ref": null,
            "rendered": "The member management tool for rural senior centers!",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Learn more",
              "className": "button large primary",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Learn more",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
